### PR TITLE
list

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: routes/routes.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+// returns a table listing all registered reports:
+exports.routes = {
+  method: 'GET',
+  path: '/reports',
```
Comment (jgallen23) [inline routes/routes.js]: this should be just `/`

File: routes/routes.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// returns a table listing all registered reports:
```
Comment (jgallen23) [inline routes/routes.js]: lets rename the file to list

File: routes/routes.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  config: {
+    plugins: {
+      'hapi-transform-table': {}
+    }
+  },
+  handler(request, h) {
+    const server = request.server;
+    const links = [];
+    Object.keys(server.methods.reports).forEach(key => {
+      const root = `${server.info.uri}/${key}`;
```
Comment (jgallen23) [inline routes/routes.js]: lets get this from the request host

File: routes/routes.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      links.push({
+        csv: `${root}.csv`,
```
Comment (jgallen23) [inline routes/routes.js]: lets add another for name that has the key

File: routes/routes.js (hunk 3 above)
Comment (jgallen23) [reply routes/routes.js]: probably needs to check for routePrefix too

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: routes/routes.js (hunk 4 above)
Comment (orthagonal) [reply routes/routes.js]: What do we need here?

Comment (jgallen23) [reply routes/routes.js]:
```
name: key,
csv: ....
```

File: routes/list.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    plugins: {
+      'hapi-transform-table': {}
+    }
+  },
+  handler(request, h) {
+    const server = request.server;
+    let base = server.settings.app.routePrefix ?
+      `${request.info.host}/${server.settings.app.routePrefix}`
+      : `${request.info.host}`;
+    base = request.url.host === 'https' ? `https://${base}` : `http://${base}`;
```
Comment (jgallen23) [inline routes/list.js]: I don't think host will ever equal https, I think you want protocol.  Probably can just remove host and just use relative paths like `/reports/blah`

**Review by jgallen23 — CHANGES_REQUESTED** [review]

**Pull request merged**